modules/secrets-store-azure.adoc

```diff
@@ -0,0 +1,112 @@
+// Module included in the following assemblies:
+//
+// * nodes/pods/nodes-pods-secrets-store.adoc
+
+:_content-type: PROCEDURE
+[id="secrets-store-azure_{context}"]
+= Mounting secrets from an external secrets store to a CSI volume
+
+TODO: intro
+
+This procedure assumes that you are using Azure Key Vault as the secrets store provider.
+
+.Prerequisites
+
+* You have installed the {secrets-store-operator}. See _Installing the Secrets Store CSI Driver_ for instructions.
+* You have installed a secrets store provider. See the appropriate vendor documentation for instructions. This procedure assumes that you have installed the link:https://azure.github.io/secrets-store-csi-driver-provider-azure/docs/getting-started/installation/[Azure Keyvault Provider].
+// TODO: Is there only a portion of these docs that they should follow? And any additional steps they need to take for OCP?
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Create a secret provider class to define your secrets store provider.
+
+.. Create a YAML file that defines the `SecretProviderClass` object:
++
+.Example `secret-provider-class.yaml`
+[source,yaml]
+----
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: my-azure-provider                 <1>
+spec:
+  provider: azure                         <2>
+  parameters:                             <3>
+    keyvaultName: "kvname"
+    objects: |
+      array:
+        - |
+          objectName: tls-cert
+          objectType: secret
+    tenantId: "tid"
+----
+<1> Specify the name for the provider secret provider class.
+<2> Specify the provider type.
+// TODO: Decide if we want to call out which are "supported"
+<3> Specify the provider-specific configuration parameters.
+
+.. Create the `SecretProviderClass` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f secret-provider-class.yaml
+----
+
+. Create a pod to use this secret provider class.
+
+.. Create a YAML file that defines the `Pod` object:
++
+.Example `pod.yaml`
+[source,yaml]
+----
+TODO: my-pod
+----
+
+.. Create the `Pod` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f pod.yaml
+----
+
+//. Get your pod to actually use it
+// .. Can show a single pod definition and apply it, but is this the best way to show it? Do people know how they need to do it?
+
+.Verification
+
+// TODO: how best to verify?
```

Comment:
@ropatil010 @knelasevero

What all should we do to verify that this was successful? The test cases didn't have these steps in them. I grabbed two examples from a demo I watched, but we'd need to adjust depending on the pod example we show above. And I wasn't sure if listing the secrets and viewing a secret were good things to verify, or if there were other things

Reply:
Currently adding tested commands.  

oc exec mydep-key-xxxx/mypod-keys  -- ls /mnt/key-store
oc exec mydep-key-xxxx/mypod-key  -- cat /mnt/key-store/key1
oc exec mydep-key-xxxx/mypod-key  -- cat /mnt/key-store/key2
oc exec mydep-key-xxxx/mypod-key  -- cat /mnt/key-store/key3

Similarly for ceritificate/secrets 
oc exec mydep-secret-xxxx/mypod-secret -- ls /mnt/secret-store
oc exec mydep-secret-xxxx/mypod-secret -- cat /mnt/secret-store/secret1

oc exec mydep-cert-xxxx/mypod-cert -- ls /mnt/cert-store
oc exec mydep-cert-xxxx/mypod-cert -- cat /mnt/cert-store/cert1

If anything needs to be added do correct me. Thanks!